apps/osvr_server.cpp

```diff
@@ -50,16 +57,84 @@ void handleShutdown() {
 int main(int argc, char *argv[]) {
     auto log = ::osvr::util::log::make_logger(OSVR_SERVER_LOG);
 
-    std::string configName(osvr::server::getDefaultConfigFilename());
-    if (argc > 1) {
-        configName = argv[1];
+    std::string configName; // server configuration filename
+
+    opt::options_description optionsAll("All Options");
+    opt::options_description optionsVisible("Command Line Options");
+    opt::positional_options_description optionsPositional;
+
+    optionsPositional.add("config", -1);
+    optionsAll.add_options()(
+        "config",
+        opt::value<std::string>(&configName)
+            ->default_value(osvr::server::getDefaultConfigFilename()),
+        "server configuration filename");
+    optionsVisible.add_options()
+        ("help", "display this help message")
+        ("verbose,v", "enable verbose logging")
+        ("debug,d", "enable debug logging");
+    optionsAll.add(optionsVisible);
+
+    opt::variables_map values;
+    try {
+        opt::store(opt::command_line_parser(argc, argv)
+                       .options(optionsAll)
+                       .positional(optionsPositional)
+                       .run(),
+                   values);
+        opt::notify(values);
+    } catch (opt::invalid_command_line_syntax &e) {
+        log->error() << e.what();
+        return 1;
+    } catch (opt::unknown_option &e) {
+        log->error() << e.what(); // may want to replace boost msg
+        return 1;
+    }
+
+    if (values.count("help")) {
+        std::cout << optionsVisible << std::endl;
+        return 0;
+    }
+
+    if (values.count("debug")) {
+        osvr::util::log::LogRegistry::instance().setLevel(osvr::util::log::LogLevel::trace);
+        osvr::util::log::LogRegistry::instance().setConsoleLevel(osvr::util::log::LogLevel::trace);
+        log->trace("Debug logging enabled.");
+    } else if (values.count("verbose")) {
+        osvr::util::log::LogRegistry::instance().setLevel(osvr::util::log::LogLevel::debug);
+        osvr::util::log::LogRegistry::instance().setConsoleLevel(osvr::util::log::LogLevel::debug);
+        log->debug("Verbose logging enabled.");
+    }
+
+    configName = values["config"].as<std::string>();
```

Comment:
@JeroMiya If no configuration file is specified, it will check to see if the file returned by `osvr::server::getDefaultConfigFilename()` exists and is readable. If it is, then we'll use it. Otherwise, we'll start the server with an empty `{ }` configuration.